src/main/java/com/google/ehub/servlets/DashboardServlet.java

```diff
@@ -49,30 +54,43 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       }
     }
 
-    SortDirection sortDir = SortDirection.valueOf(sortingDirection);
-    EntertainmentItemList itemList;
-
-    if (searchValue.isEmpty()) {
-      itemList = EntertainmentItemDatastore.getInstance().queryAllItemsWithTitleOrder(
-          fetchOptions, sortDir);
-    } else {
-      itemList = EntertainmentItemDatastore.getInstance().queryItemsByTitlePrefix(
-          fetchOptions, searchValue, sortDir);
-    }
-
     response.setContentType("application/json");
-    response.getWriter().println(new Gson().toJson(itemList));
+    response.getWriter().println(
+        new Gson().toJson(getItemList(fetchOptions, searchValue, sortType)));
   }
 
   /**
    * Verifies if the parameters given in the HTTP Get Request are valid.
    *
    * @param searchValue the search value given in the Get request parameter
-   * @param sortingDirection the sorting direction given in the Get request parameter
-   * @return true if the parameters given in the Get request are valid, false otherwise
+   * @param sortType the sort type given in the Get request parameter
+   * @return true if the parameters given in the Get request are valid, false
+   *     otherwise
    */
-  private static boolean areGetRequestParametersValid(String searchValue, String sortingDirection) {
-    return (searchValue != null && searchValue.length() <= MAX_SEARCH_VALUE_CHARS)
-        && sortingDirection != null && EnumUtils.isValidEnum(SortDirection.class, sortingDirection);
+  private static boolean areGetRequestParametersValid(String searchValue,
+                                                      String sortType) {
+    return (searchValue != null &&
+            searchValue.length() <= MAX_SEARCH_VALUE_CHARS) &&
+        sortType != null &&
+        (sortType.equals(ASCENDING_TITLE_SORT) ||
+         sortType.equals(DESCENDING_TITLE_SORT) ||
+         sortType.equals(RECENT_RELEASE_DATE_SORT));
+  }
+
+  private static EntertainmentItemList
+  getItemList(FetchOptions fetchOptions, String searchValue, String sortType) {
```

Comment:
spacing here looks off

Reply:
Yeah I tried formatting a single file using ```clang-format -i <filename>``` but that seems to use a different kind of formatting style.